components/brave_wallet_ui/utils/tx-utils.ts

```diff
@@ -1051,14 +1051,15 @@ export const isSendingToKnownTokenContractAddress = (
  * @param tx - The transaction to check
  */
 export const transactionHasSameAddressError = (
-  tx: TransactionInfo
+  tx: TransactionInfo,
+  txAccount: BraveWallet.AccountInfo
```

Comment:
Maybe we can also derive txAccount from TransactionInfo here?

Reply:
I believe we'll need to perform a `useAccountQuery` for that, which is not possible outside the context of  react hook.